src/memfd.rs

```diff
@@ -233,3 +248,27 @@ impl FromRawFd for Memfd {
         Memfd { file }
     }
 }
+
+impl AsRawFd for Memfd {
+    fn as_raw_fd(&self) -> RawFd {
+        self.file.as_raw_fd()
+    }
+}
+
+impl IntoRawFd for Memfd {
+    fn into_raw_fd(self) -> RawFd {
+        self.into_file().into_raw_fd()
+    }
+}
+
+/// Check if a file descriptor is a memfd.
+///
+/// Implemented by trying to retrieve the seals.
+/// If that fails, the fd is not a memfd.
+fn is_memfd<F: AsRawFd>(fd: &F) -> bool {
```

Comment:
Nit: I don't see much purpose in keeping this generic and would move the `as_raw_fd()` calls to the caller.

Reply:
There is a very benefit in my opinion. If we move the call to the caller, we're taking a `RawFd`, and the function must be unsafe. If we take an `AsRawFd`, the trait guarantees that `fd` is a valid file descriptor.

It's only an internal API, so it doesn't matter too much either way. But I would keep it this way so we know for sure the file descriptor is valid.